### DISCUSSION TITLE
Minecraft 中打造出一台完整電腦

### DISCUSSION THREAD

**Discussion by HSU-YU-JEN** [1101-GS4719-C/Python-Community-GS4719-C#244] [科技新知]
有強者玩家花了 7 個月在 Minecraft 打造出「一台電腦」，是世界中至今最大且最複雜的 CPU，就連經典遊戲《貪食蛇》、《俄羅斯方塊》都能遊玩!!!!!!
影片連結如下
https://www.youtube.com/watch?v=FDiapbD0Xfg

**Comment by HSU-YU-JEN** [comment]
這台電腦「Chungus 2」由 Minecraft 玩家 Sammyuri 所打造，是一台極其複雜的電腦處理器，模擬了一個 8 位元處理器，外媒《PCWorld》指出，這個長達數月的 project 「展示了 CPU 背後的物理原理」，雖然是模擬的，但還是能讓我們能一窺電腦晶片的樣貌。

**Comment by cody910120** [comment]
我只能說這位大神用電腦創造電腦，牛的一匹

**Comment by Isongzhe** [comment]
我之前也有在巴哈看到，跟鬼一樣

**Comment by luo910227** [comment]
用這樣的方式展示 CPU 背後的物理原理，著實偉大，這樣對於喜歡Minecraft 的人是著實有教育意義的

**Comment by YuHengChen54** [comment]
這真的太扯了，我連Minecraft裡面最基礎的指令怎麼打都不會XD

**Comment by enoch-lightning** [comment]
我看了一下影片底下評論區的留言，有不少人說這個東西其實沒有那麼難。我還沒有很理解作者是怎麼做的。我覺得比較重要的是，作者好像也是用python寫的。寫法有部份是組合語言(assembly language)。有人說作者連電晶體都用minecraft製作，這我就不敢肯定，因為我不相信那個人真的有把作者的程式從頭到尾看過。如果真的要達到CPU的功能，單以寫程式的話，可能不需要懂太多物理就可以達成。只要最後有一樣或類似的功能就好，條條大路通羅馬。

**Reply by enoch-lightning** [reply]
以下是作者此minecraft專案的儲存庫截圖。
![image](https://user-images.githubusercontent.com/86505454/149117947-3ef4109a-4f31-4b0f-8b6d-509f1d289e60.png)
